runelite-client/src/main/java/net/runelite/client/plugins/runecraft/RunecraftPlugin.java

```diff
@@ -133,36 +138,41 @@ public void updateConfig(ConfigChanged event)
 	@Subscribe
 	public void onChatMessage(ChatMessage event)
 	{
-		if (event.getType() != ChatMessageType.SERVER || !config.showBindNeck())
+		if (event.getType() != ChatMessageType.SERVER && config.showBindNeck())
 		{
-			return;
-		}
+			Matcher match = bindNeckString.matcher(event.getMessage());
+			if (match.find())
+			{
+				bindNeckOverlay.bindingCharges = Integer.parseInt(match.group(1));
+				return;
+			}
 
-		Matcher match = bindNeckString.matcher(event.getMessage());
-		if (match.find())
-		{
-			bindNeckOverlay.bindingCharges = Integer.parseInt(match.group(1));
-			return;
-		}
+			if (event.getMessage().contains("You bind the temple's power"))
+			{
+				if (event.getMessage().contains("mud")
+					|| event.getMessage().contains("lava")
+					|| event.getMessage().contains("steam")
+					|| event.getMessage().contains("dust")
+					|| event.getMessage().contains("smoke")
+					|| event.getMessage().contains("mist"))
+				{
+					bindNeckOverlay.bindingCharges -= 1;
+					return;
+				}
+			}
 
-		if (event.getMessage().contains("You bind the temple's power"))
-		{
-			if (event.getMessage().contains("mud")
-				|| event.getMessage().contains("lava")
-				|| event.getMessage().contains("steam")
-				|| event.getMessage().contains("dust")
-				|| event.getMessage().contains("smoke")
-				|| event.getMessage().contains("mist"))
+			if (event.getMessage().contains("Your Binding necklace has disintegrated."))
 			{
-				bindNeckOverlay.bindingCharges -= 1;
-				return;
+				//set it to 17 because this message is triggered first before the above chat event
+				bindNeckOverlay.bindingCharges = 17;
 			}
 		}
-
-		if (event.getMessage().contains("Your Binding necklace has disintegrated."))
+		if (config.degradingNotification())
 		{
-			//set it to 17 because this message is triggered first before the above chat event
-			bindNeckOverlay.bindingCharges = 17;
+			if (event.getMessage().contains("Your pouch has decayed through use."))
+			{
+				notifier.notify(POUCH_DECAYED_MESSAGE);
+			}
```

Comment:
There should be a `return` statement after the above line.

Reply:
Done